api/src/main/java/org/pmiops/workbench/db/dao/UserServiceImpl.java

```diff
@@ -883,6 +885,35 @@ public DbUser syncTwoFactorAuthStatus(DbUser targetUser, Agent agent, boolean is
         agent);
   }
 
+  @Override
+  public DbUser syncDuccVersionStatus(
+      DbUser targetUser, Agent agent, @Nullable Integer signedDuccVersion) {
+    if (isServiceAccount(targetUser)) {
+      // Skip sync for service account user rows.
+      return targetUser;
+    }
+
+    return updateUserWithRetries(
+        user -> {
+          // convert Integer to int to prevent a NPE from comparison-unboxing
+          final int signedVersionForComparison =
+              Optional.ofNullable(signedDuccVersion)
+                  // null is invalid, so convert to a known-invalid int
+                  .orElse(CURRENT_DATA_USER_CODE_OF_CONDUCT_VERSION - 1);
+
+          if (signedVersionForComparison != getCurrentDuccVersion()) {
```

Comment:
nit: here we use the method, then 2 lines above we use the constant - would use one or the other in both places for consistency